3p/iframe-messaging-client.js

```diff
@@ -116,7 +125,20 @@ export class IframeMessagingClient {
   }
 
   /**
-   * @param {!Window} win
+   * Determines whether frameOrWindow is a frame, or a window.
+   * @returns {boolean}
+   */
+  hostWindowIsActuallyAnIframe_() {
+    // If it's a window, it will have .postMessage
+    // We check for that before .contentWindow, since a cross-domain window
+    // may throw if we try to access anything unsafe
+    return (!this.hostWindow_.postMessage && !!this.hostWindow_.contentWindow);
```

Comment:
That's not what I'm doing here. I'm testing whether this.hostWindow_ is actually a window, or an iframe. If it has .postMessage then it's a window.